src/main/java/com/example/booking_app/service/SeatsServiceImplementation/CinemaSeatServiceImplementation.java

```diff
@@ -0,0 +1,233 @@
+                    }
+                }
+             if (canBeReserved.contains(false)) {
+                blankBetweenReservedFields = true;
+            } else {
+                blankBetweenReservedFields = false;
+
+            }
+        }
+
+        return blankBetweenReservedFields;
+    }
+    @Override
+    public String printReservationStatus(String firstName, String lastName, List <Long> listOfSeats, int adult, int student, int child){
+        double price = adult * (ETicketType.TICKET_ADULT.getPrice()) + student * (ETicketType.TICKET_STUDENT.getPrice()) + child * (ETicketType.TICKET_CHILD.getPrice());
+        List<Boolean> isReserved = new ArrayList<>();
+        List<Boolean> canBeReserved = new ArrayList<>();
+        boolean blankBetweenReservedFields = true;
+
+        List<CinemaSeat> seats = new ArrayList<>();
+
+        for (Long id : listOfSeats) {
+            CinemaSeat seat = this.read(id);
+            seats.add(seat);
+        }
+        for (CinemaSeat cinemaSeat : seats) {
+            if (cinemaSeat.getReservation() == null) {
+                isReserved.add(true);
+            } else {
+                isReserved.add(false);
+            }
+        }
+        if (isReserved.contains(false)) {
+            return "One of the seats is reserved";
+        } else {
+            blankBetweenReservedFields = this.checkIfBlankBetweenReservedSeats(seats);
+
+            for (CinemaSeat cinemaSeat : seats) {
+                if (LocalDateTime.now().plusMinutes(15).isAfter(cinemaSeat.getScreening().getDate())) {
+                    canBeReserved.add(false);
+                } else {
+                    canBeReserved.add(true);
+                }
+            }
+            if (canBeReserved.contains(false)) {
+                return "You can't book seats. Seats can be booked at latest 15 minutes before the screening begins.";
+            } else if(blankBetweenReservedFields==true){
+                return "You can't book seats. There cannot be a single place left over in a row between two already reserved\n" +
+                        "places.";
+            } else {
+                for (CinemaSeat seat : seats) {
+                    Reservation reservation = new Reservation();
+                    reservation.setExirationDate(LocalDateTime.now().plusMinutes(15));
+                    reservation.setCinemaSeat(seat);
+                    reservation.setFirstName(firstName);
+                    reservation.setLastName(lastName);
+                    reservationService.create(reservation);
+                    seat.setReservation(reservation);
+                    this.update(seat.getId(), seat);
+                }
+                return "Total amount to pay: " + price + "PLN" + "\nExpiration Time: " + LocalTime.now().plusMinutes(15).toString();
+            }
+        }
+    }
+
+@Override
+    public boolean checkIfSeatIsReserved(CinemaSeat seat){
+        if (seat.getReservation() == null) {
```

Comment:
`return seat.getReservation() != null`? W ogóle to funkcja `isReserved` mogłaby być w `CinemaSeat`.

Reply:
Zgadzam się. Powyższa metoda jest ściśle związana z klasą CinemaSeat i mogłaby być jej częścią.